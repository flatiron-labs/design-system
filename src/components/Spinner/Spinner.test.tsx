import React from 'react'
import { render, screen } from '~/test/utils'
import { Spinner, SpinnerProps } from '.'

describe('SideNav', () => {
  const altText = 'loading indicator'

  const renderComponent = (props?: SpinnerProps) => {
    render(<Spinner {...props} />)
  }

  it('should render the component', () => {
    renderComponent()
    screen.getByAltText(altText)
  })

  it('should render with alt text', () => {
    const alt = 'Data loading indicator'
    renderComponent({ alt })
    expect(screen.getByAltText(alt)).toMatchSnapshot()
  })

  it('should render with styles', () => {
    renderComponent({ style: { padding: '10px' } })
    expect(screen.getByAltText(altText)).toMatchSnapshot()
  })
})
