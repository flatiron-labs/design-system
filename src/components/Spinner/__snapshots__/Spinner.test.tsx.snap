// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should render with alt text 1`] = `
<img
  alt="Data loading indicator"
  height="80"
  src=""
  width="80"
/>
`;

exports[`SideNav should render with styles 1`] = `
<img
  alt="loading indicator"
  height="80"
  src=""
  style="padding: 10px;"
  width="80"
/>
`;
